src/java.base/share/classes/java/util/Arrays.java

```diff
@@ -77,6 +79,27 @@ public final class Arrays {
     // Suppresses default constructor, ensuring non-instantiability.
     private Arrays() {}
 
+    /**
+     * Sorts the specified array into ascending numerical order.
+     *
+     *
+     * @param elemType the class of the array to be sorted
+     * @param array the array to be sorted
+     * @param offset the relative offset, in bytes, from the base address of
+     * the array to sort, otherwise if the array is {@code null},an absolute
+     * address pointing to the first element to sort from.
+     * @param fromIndex the index of the first element, inclusive, to be sorted
+     * @param toIndex the index of the last element, exclusive, to be sorted
+     */
+    @IntrinsicCandidate
+    public static void arraySort(Class<?> elemType, Object array, long offset, int fromIndex, int toIndex) {
+        if (elemType == int.class) DualPivotQuicksort.sort((int[]) array, 0, fromIndex, toIndex);
+        else if (elemType == long.class) DualPivotQuicksort.sort((long[]) array, 0, fromIndex, toIndex);
+        else if (elemType == float.class) DualPivotQuicksort.sort((float[]) array, 0, fromIndex, toIndex);
+        else if (elemType == double.class) DualPivotQuicksort.sort((double[]) array, 0, fromIndex, toIndex);
+        else throw new UnsupportedOperationException("arraySort intrinsic not supported for this type: " + elemType.toString());
```

Comment:
If-else chain can cause performance degradation on targets which do not currently intrinsify arraySort.